### PR TITLE
better handling of float ops in wasm2asm

### PR DESCRIPTION
**Pull request by froydnj**
Running `wasm2asm` on files in `test/` revealed some errors in handling float operations.  This pull request is a first attempt at handling them.  More complex operations, like conversion, reinterpretation, and copysign, are left for later.

### CODE REVIEW

**Comment by kripken** [comment]
These changes don't require updates in the existing wasm2asm tests?

**Comment by froydnj** [comment]
> These changes don't require updates in the existing wasm2asm tests?

Good point, added tests for translating everything (no assertions yet, though).  The asm.js looks a little funny (multiple returns, for instance), but that can be handled in a followup.

File: test/float-ops.2asm.js [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+ var HEAPF64 = new global.Float64Array(buffer);
+ var Math_imul = global.Math.imul;
+ var Math_fround = global.Math.fround;
+ var Math_abs = global.Math.abs;
+ var Math_clz32 = global.Math.clz32;
+ function dummy() {
+  
+ }
+ 
+ function $$1($$0, $$1) {
```
Comment (kripken) [inline test/float-ops.2asm.js]: btw, it would be nice to have wasm2asm emit proper function names (same name as in the wasm)

**Comment by kripken** [comment]
Looks good, thanks!

File: test/float-ops.2asm.js (hunk 1 above)
Comment (froydnj) [reply test/float-ops.2asm.js]: For ease of debugging?

Comment (kripken) [reply test/float-ops.2asm.js]: Yeah, I think it would help debugging and also reading the testcases.

Comment (kripken) [reply test/float-ops.2asm.js]: Maybe we should only do it with `-g` or something like that?

**Pull request merged**